### PR TITLE
raft: add comments for raft storage

### PR DESCRIPTION
**Pull request by hhkbp2**
Fix issue #1051

### CODE REVIEW

File: src/raft/storage.rs [hunk 1]
```diff
@@ -38,19 +38,43 @@ pub struct RaftState {
     pub conf_state: ConfState,
 }
 
+/// Storage is an trait that may be implemented by the application
+/// to retrieve log entries from storage.
+///
+/// If any Storage metho returns an error, the raft instance will
```
Comment (BusyJay) [inline src/raft/storage.rs]: s/will/may/
s/metho/method/

File: src/raft/storage.rs [hunk 2]
```diff
@@ -66,6 +90,7 @@ impl Default for MemStorageCore {
 }
 
 impl MemStorageCore {
+    // set_hardstate saves the current HardState.
```
Comment (BusyJay) [inline src/raft/storage.rs]: Use `///` for docs comment. See also https://doc.rust-lang.org/stable/book/documentation.html

File: src/raft/storage.rs (hunk 1 above)
Comment (hhkbp2) [inline src/raft/storage.rs]: will is original word of etcd/raft.
Address comment.

**Comment by BusyJay** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**